src/Illuminate/Foundation/Testing/DatabaseTruncates.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Illuminate\Foundation\Testing;
+
+use Illuminate\Foundation\Testing\Traits\CanConfigureMigrationCommands;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+trait DatabaseTruncates
+{
+    use CanConfigureMigrationCommands;
+
+    protected static array $allTables;
+
+    protected function truncateTables()
+    {
+        // Always remove any test data before the application is destroyed.
+        $this->beforeApplicationDestroyed(function () {
+            Schema::disableForeignKeyConstraints();
+            collect(static::$allTables ??= DB::connection()->getDoctrineSchemaManager()->listTableNames())
+                ->diff($this->excludeTables())
+                ->filter(fn ($table) => DB::table($table)->exists())
+                ->each(fn ($table) => DB::table($table)->truncate());
+        });
+
+        // Migrate and seed the database on first run.
+        if (! RefreshDatabaseState::$migrated) {
+            $this->artisan('migrate:fresh', $this->migrateFreshUsing());
+            RefreshDatabaseState::$migrated = true;
+
+            return;
+        }
+
+        // Seed the database on subsequent runs.
+        $this->artisan('db:seed', ['--class' => $this->seeder()]);
```

Comment:
What if someone doesn't want to seed?

Reply:
I've added a check around the `seeder()`, and `shouldSeeder()` methods.